beacon-chain/sync/validate_aggregate_proof.go

```diff
@@ -54,6 +54,10 @@ func (s *Service) validateAggregateAndProof(ctx context.Context, pid peer.ID, ms
 	if s.hasSeenAggregatorIndexEpoch(m.Message.Aggregate.Data.Target.Epoch, m.Message.AggregatorIndex) {
 		return pubsub.ValidationIgnore
 	}
+	// Check that the block being voted on isn't invalid.
+	if s.hasBadBlock(bytesutil.ToBytes32(m.Message.Aggregate.Data.BeaconBlockRoot)) {
```

Comment:
Should we check target root too?

Reply:
Yeah that makes sense too, although it isnt part of the spec. Would be a logical check to add too over here.